typegen/src/typegen/error.rs

```diff
@@ -35,6 +35,16 @@ pub enum TypegenError {
     /// Use [`crate::utils::ensure_unique_type_paths`] on your [`scale_info::PortableRegistry`] to deduplicate type paths.
     #[error("There are two types with the the same type path {0} but different structure. Use `scale_typegen::utils::ensure_unique_type_paths` on your `PortableRegistry` before, to avoid this error.")]
     DuplicateTypePath(String),
+    /// PortableRegistry entry has incorrect Id.
+    #[error("PortableRegistry entry has incorrect type_id. expected type_id: {expected_ty_id}, got: {given_ty_id}.\nDefinition of the type: {ty_def}.\nThis can happen if registry was modified with calls to `::retain()` in older versions of scale-info. Try generating a new metadata for use with `TypeGenerator`.")]
+    RegistryIncorrect {
```

Comment:
Nit: I might call this something like `RegistryTypeIdsInvalid` or something closer to the specific issue, but not too bothered!

Reply:
InvalidRegistry would be my choice but this is ok for me as well

Reply:
renamed to `RegistryTypeIdsInvalid`